setup.py

```diff
@@ -786,20 +451,44 @@ def load_config(config_path: Path) -> Dict[str, str]:
     return parse_config_vars(config_data)
 
 
-def _build_spec(preset_name: str, fork: str,
-                source_files: Sequence[Path], preset_files: Sequence[Path], config_file: Path) -> str:
+def build_spec(preset_name: str, fork: str, source_files: Sequence[Path], preset_files: Sequence[Path],
+               config_file: Path, spec_dir: Path, spec_yaml: Dict) -> str:
     preset = load_preset(preset_files)
     config = load_config(config_file)
     all_specs = [get_spec(spec, preset, config) for spec in source_files]
 
-    spec_object = all_specs[0]
-    for value in all_specs[1:]:
+    python_prefix = []
+    for path in spec_yaml.get("py_prefix", []):
+        with open(os.path.join(spec_dir, path)) as file:
+            python_prefix.append(file.read())
+
+    python_suffix = []
+    for path in spec_yaml.get("py_suffix", []):
+        with open(os.path.join(spec_dir, path)) as file:
+            python_suffix.append(file.read())
+
+    spec_object = SpecObject(
+        functions={},
+        protocols={},
+        custom_types={},
+        vars={},
+        config_vars={},
+        ssz_objects={},
+        dataclasses={},
+        python_prefix=python_prefix,
+        python_suffix=python_suffix,
+    )
+    for value in all_specs:
         spec_object = combine_spec_objects(spec_object, value)
 
+    for function in spec_yaml.get("overridden_functions", []):
+        if function in spec_object.functions:
+            del spec_object.functions[function]
```

Comment:
IIRC, it seems that `ceillog2` and `floorlog2` do not need to be overridden?

Reply:
You are correct. The original `setup.py` contained overriding logic that I ported over without checking.